packages/backend-common/src/database/types.ts

```diff
@@ -30,6 +30,14 @@ export interface PluginDatabaseManager {
    * stores so that plugins are discouraged from database integration.
    */
   getClient(): Promise<Knex>;
+
+  /**
+   * runMigrations can be used to determine if database migrations
+   * should be performed.
+   *
+   * Useful if connecting to a read-only database.
+   */
+  runMigrations: boolean;
```

Comment:
I'm feeling we can future proof this a little bit more. I think there's space for a little bit of an options category of `migrations`, so the entire thing ends up somewhere here?

```ts
DatabaseManager.fromConfig(config, {
  migrations: {
    apply: false,
  }
})

// ...

if (db.migrations.apply) {
  applyMigrations(...)
}
```

Reply:
> I think there's space for a little bit of an options category of `migrations`

@Rugvip Makes sense 👍  

Still in progress, but I've made some changes. I made the `options` argument optional as well as the `migrations` object in it. Which had the effect of using it like 
```ts
if (database.migrations?.apply) {
  ...
}
```
Not sure if that's what we want? The alternative is to make that property required if you pass `options` which will require a few more changes, mostly to tests and `standaloneServer`. WDYT?